robot/framesystem/framesystem.go

```diff
@@ -31,16 +31,48 @@ var API = resource.APINamespaceRDKInternal.WithServiceType(SubtypeName)
 var InternalServiceName = resource.NewName(API, "builtin")
 
 // A Service that returns the frame system for a robot.
+//
+// TransformPose example:
+//
+//	// Define a Pose coincident with the world reference frame
+//	firstPose := spatialmath.NewPoseFromPoint(r3.Vector{X: 0.0, Y: 0.0, Z: 0.0})
+//
+//	// Establish the world as the reference for firstPose
+//	firstPoseInFrame := referenceframe.NewPoseInFrame(referenceframe.World, firstPose)
+//
+//	// Calculate firstPoseInFrame from the perspective of the origin frame of myArm
+//	transformedPoseInFrame, err := machine.TransformPose(context.Background(), firstPoseInFrame, "myArm", nil)
+//
+// TransformPointCloud example:
+//
+//	// Create an empty slice to store point cloud data.
+//	pointClouds := make([]pointcloud.PointCloud, 0)
+//
+//	Transform the first point cloud in the list from its reference frame to the frame of 'myArm'.
+//	transformed, err := machine.TransformPointCloud(context.Background(), pointClouds[0], referenceframe.World, "myArm")
+//
+// CurrentInputs example:
+//
+//	myCurrentInputs, err := myArm.CurrentInputs(context.Background())
 type Service interface {
 	resource.Resource
+
+	// TransformPose returns a transformed pose in the destination reference frame.
+	// This method converts a given source pose from one reference frame to a specified destination frame.
 	TransformPose(
 		ctx context.Context,
 		pose *referenceframe.PoseInFrame,
 		dst string,
 		additionalTransforms []*referenceframe.LinkInFrame,
 	) (*referenceframe.PoseInFrame, error)
+
+	// TransformPointCloud returns a new point cloud where the points have been adjusted from one reference frame to a specified destination frame.
 	TransformPointCloud(ctx context.Context, srcpc pointcloud.PointCloud, srcName, dstName string) (pointcloud.PointCloud, error)
+
+	// CurrentInputs returns a map of the current inputs for a machine's frame system and a map of statuses indicating whether these inputs are active.
 	CurrentInputs(ctx context.Context) (map[string][]referenceframe.Input, map[string]referenceframe.InputEnabled, error)
+
+	// FrameSystem returns the frame system of the machine and incorporates any specified additional transformations.
 	FrameSystem(ctx context.Context, additionalTransforms []*referenceframe.LinkInFrame) (referenceframe.FrameSystem, error)
```

Comment:
An example of this could be
```
frameSystem, err := fsService.FrameSystem(context.Background(), nil)
```
Here we just return the framesystem with no additional transforms, i.e. this returns the framesystem the user built using app.viam